driver/bpf/fillers.h

```diff
@@ -5752,4 +5753,701 @@ FILLER(sys_dup3_x, true)
+	if(res != PPM_SUCCESS)
+	{
+		return res;
+	}
+
+	/* Parameter 12: vm_rss (type: PT_UINT32) */
+	res = bpf_val_to_ring_type(data, total_rss, PT_UINT32);
+	if(res != PPM_SUCCESS)
+	{
+		return res;
+	}
+
+	/* Parameter 13: vm_swap (type: PT_UINT32) */
+	res = bpf_val_to_ring_type(data, swap, PT_UINT32);
+	if(res != PPM_SUCCESS)
+	{
+		return res;
+	}
+
+	/* Parameter 14: comm (type: PT_CHARBUF) */
+	res = bpf_val_to_ring_type(data, (unsigned long)child->comm, PT_CHARBUF);
+	if(res != PPM_SUCCESS)
+	{
+		return res;
+	}
+
+	bpf_tail_call(data->ctx, &tail_map, PPM_FILLER_sched_prog_fork_2);
+	bpf_printk("Can't tail call 'sched_prog_fork_2' filler\n");
+	return PPM_FAILURE_BUG;
+}
+
+FILLER(sched_prog_fork_2, false)
+{
+	int res = 0;
+	int cgroups_len = 0;
+	struct sched_process_fork_raw_args* original_ctx = (struct sched_process_fork_raw_args*)data->ctx;
+	struct task_struct *child = (struct task_struct *)original_ctx->child;
+
+	res = bpf_append_cgroup(child, data->tmp_scratch, &cgroups_len);
+	if(res != PPM_SUCCESS)
+	{
+		return res;
+	}
+
+	/* Parameter 15: cgroups (type: PT_CHARBUFARRAY) */
+	res = __bpf_val_to_ring(data, (unsigned long)data->tmp_scratch, cgroups_len, PT_BYTEBUF, -1, false);
+	if(res != PPM_SUCCESS)
+	{
+		return res;
+	}
+
+	bpf_tail_call(data->ctx, &tail_map, PPM_FILLER_sched_prog_fork_3);
+	bpf_printk("Can't tail call 'sched_prog_fork_3' filler\n");
+	return PPM_FAILURE_BUG;
+}
+
+FILLER(sched_prog_fork_3, false)
+{
+	int res = 0;
+	struct sched_process_fork_raw_args* original_ctx = (struct sched_process_fork_raw_args*)data->ctx;
+	struct task_struct *child = (struct task_struct *)original_ctx->child;
+	struct task_struct *parent = (struct task_struct *)original_ctx->parent;
+	uint32_t flags = 0;
+
+	/* Since Linux 2.5.35, the flags mask must also include
+	 * CLONE_SIGHAND if CLONE_THREAD is specified (and note that,
+	 * since Linux 2.6.0, CLONE_SIGHAND also requires CLONE_VM to
+	 * be included). 
+	 * Taken from https://man7.org/linux/man-pages/man2/clone.2.html
+	 */
+	pid_t tid = _READ(child->pid);
+	pid_t tgid = _READ(child->tgid);
+	if(tid != tgid)
+	{
+		flags |= PPM_CL_CLONE_THREAD | PPM_CL_CLONE_SIGHAND | PPM_CL_CLONE_VM;
+	}
+	
+	/* If CLONE_FILES is set, the calling process and the child
+	 * process share the same file descriptor table.
+	 * Taken from https://man7.org/linux/man-pages/man2/clone.2.html
+	 */
+	struct files_struct * file_struct = NULL;
+	struct files_struct * parent_file_struct = NULL;
+	file_struct = _READ(child->files);
+	parent_file_struct = _READ(parent->files);
+	if(parent_file_struct == file_struct)
+	{
+		flags |= PPM_CL_CLONE_FILES;
+	}
+
+	/* It's possible to have a process in a PID namespace that 
+	 * nevertheless has tid == vtid,  so we need to generate this
+	 * custom flag `PPM_CL_CHILD_IN_PIDNS`.
+	 */
+	struct pid_namespace *pidns = bpf_task_active_pid_ns(child);
+	int pidns_level = _READ(pidns->level);
+	if(pidns_level != 0)
+	{
+		flags |= PPM_CL_CHILD_IN_PIDNS;
+	}
```

Comment:
In the end, we want to detect every child that is spawned in a PID namespace different from the init one so we can keep only this logic in the `if` statement